### DISCUSSION TITLE
Should NodeJS session.load<TypeA>(id, TypeA) return TypeB?

### DISCUSSION THREAD

**Discussion by Bushuo** [ravendb/ravendb#14343] [Q&A] [answered]
I want to understand the difference between load and query.

Suppose I have two documents of different Type and Collection
```ts
{
    '@metadata': {
        'Raven-Node-Type': 'TypeA',
        '@collection': 'TypeA',
        '@id': 'type-a-id'
    }
}

{
   '@metadata': {
        'Raven-Node-Type': 'TypeB',
         '@collection': 'TypeB',
        '@id': 'type-b-id'
    }
}
```
If I want to load a document of TypeA with 
```ts
session.load<TypeA>(id, TypeA)
```
but `id="type-b-id"`, should the TypeB document be returned?

**Comment by ayende** [comment]
In node.js, there are not actual types. If you are giving the wrong value, it will load the wrong item, yes.

**Comment by ml054** [comment, marked as answer]
hi @Bushuo 

This was a design decision to not block it by default and one of breaking changes between 3.x and 4.x product. In C# Client API you can still hook to the OnBeforeConversionToEntity or OnAfterConversionToEntity events (https://ravendb.net/docs/article-page/5.3/csharp/client-api/session/how-to/subscribe-to-events#onbeforeconversiontoentity) and throw there is this is the case. 

The events are exposed in node.js client, you can find usage example here: https://github.com/ravendb/ravendb-nodejs-client/blob/7b1fa6447880225924d1ea05e756e4d5344540ce/test/Ported/Issues/RavenDB_9889.ts#L69

**Reply by Bushuo** [reply]
Thank you for the insight :) Hooking into the event seems to be a great way to solve my problem. 
I will try that.
